src/nile/cli.py

```diff
@@ -84,15 +84,47 @@ def run(path, network):
 
 
 @cli.command()
-@click.argument("artifact", nargs=1)
-@click.argument("arguments", nargs=-1)
-@network_option
+@click.argument("contract_name", nargs=1)
+@click.argument("params", nargs=-1)
+@click.option("--max_fee", nargs=1)
+@click.option("--salt", nargs=1, default=0)
+@click.option("--unique", is_flag=True)
+@click.option("--account")
 @click.option("--alias")
 @click.option("--abi")
+@click.option("--udc_address")
 @mainnet_token_option
-def deploy(artifact, arguments, network, alias, abi=None, token=None):
+@network_option
+def deploy(
+    contract_name,
+    salt,
+    params,
+    max_fee,
+    unique,
+    account,
+    alias,
+    abi,
+    udc_address,
+    token,
+    network,
+):
     """Deploy StarkNet smart contract."""
-    deploy_command(artifact, arguments, network, alias, abi=abi, mainnet_token=token)
+    if account is not None:
+        account = Account(account, network)
+        account.deploy_contract(
+            contract_name,
+            salt,
+            unique,
+            params,
+            alias,
+            deployer_address=udc_address,
+            max_fee=max_fee,
+            abi=abi,
```

Comment:
no `mainnet_token`?

Reply:
I assumed that we shouldn't need a mainnet token to deploy through an invoke transaction as we are doing here, but I must have confirmed this with starkware team. Let me follow up on slack with them.

Reply:
From the answer, i think I assumed correctly.